src/aggregator/client/queue_test.go

```diff
@@ -43,17 +43,23 @@ func TestInstanceQueueEnqueueClosed(t *testing.T) {
 }
 
 func TestInstanceQueueEnqueueQueueFullDropCurrent(t *testing.T) {
-	opts := testOptions().SetInstanceQueueSize(1)
+	opts := testOptions().
+		SetInstanceQueueSize(1).
+		SetQueueDropType(DropCurrent).
+		SetMaxBatchSize(0)
 	queue := newInstanceQueue(testPlacementInstance, opts).(*queue)
-	queue.dropType = DropCurrent
 
 	// Fill up the queue and park the draining goroutine so the queue remains full.
 	queue.writeFn = func([]byte) error {
 		select {}
 	}
-	queue.bufCh <- testNewBuffer(nil)
-	queue.bufCh <- testNewBuffer(nil)
-	require.Equal(t, errWriterQueueFull, queue.Enqueue(testNewBuffer(nil)))
+
+	time.Sleep(5 * time.Millisecond)
```

Comment:
Was this test flaky before? Would really like to avoid having tests that depends on sleeps

Reply:
yeah it was super flaky, I can just rework it